src/pinns_pde_solve.jl

```diff
@@ -1,5 +1,55 @@
-struct PhysicsInformedNN{D}
+struct PhysicsInformedNN{D,C,P,PH,DER,T,K}
   dx::D
+  chain::C
+  initθ::P
+  phi::PH
+  autodiff::Bool
+  derivative::DER
+  training_strategies::T
+  kwargs::K
+end
+
+function PhysicsInformedNN(dx,
+                           chain,
+                           init_params = nothing;
+                           _phi = nothing,
+                           autodiff=false,
+                           _derivative = nothing,
+                           training_strategies = TrainingStrategies(),
```

Comment:
I think you just want this to just be like `training_strategy = StochasticSampling()` and stuff like that, so you can have training-mode-specific caches and arguments.

Reply:
I think that can be handled in a next PR though. This has a lot in it, looks great, so I'd like to merge and make the next diff clearer.